silkworm/capi/silkworm.cpp

```diff
@@ -424,6 +419,24 @@ SILKWORM_EXPORT int silkworm_add_snapshot(SilkwormHandle handle, SilkwormChainSn
     return SILKWORM_OK;
 }
 
+SILKWORM_EXPORT int silkworm_add_state_snapshot(SilkwormHandle handle, const SilkwormStateSnapshot* snapshot) SILKWORM_NOEXCEPT {
```

Comment:
consider moving this implementation (and also silkworm_add_snapshot implementation) from capi/silkworm.cpp to the capi/snapshots.cpp separate file. we do this for other calls, see: silkworm_sentry_start, silkworm_start_fork_validator